src/shared/sam/cli/samCliConfiguration.ts

```diff
@@ -40,23 +53,26 @@ export class DefaultSamCliConfiguration implements SamCliConfiguration {
         )
     }
 
-    /**
-     * Initializes this SamCliConfiguration object from the VSCode user settings,
-     * or tries to auto-detect `sam` in the environment.
-     */
-    public async initialize(): Promise<void> {
-        const configLocation = this.getSamCliLocation() ?? ''
-        if (configLocation) {
-            if (await filesystemUtilities.fileExists(configLocation)) {
-                return
-            }
+    public async getOrDetectSamCli(): Promise<{ path: string; autoDetected: boolean }> {
+        const fromConfig =
+            this._configuration.readSetting<string>(DefaultSamCliConfiguration.CONFIGURATION_KEY_SAMCLI_LOCATION) ?? ''
+        // Respect user setting, do not validate it. fileExists() does not
+        // understand WSL paths, for example.  https://github.com/aws/aws-toolkit-vscode/issues/1300
+        if (fromConfig) {
```

Comment:
Key change: instead of checking fileExists() here, just check if the user has set the setting.